### ISSUE TITLE
transmute_int_to_float false positive

### ISSUE THREAD

**Issue by leonardo-m** [rust-lang/rust-clippy#5747]
This code:
```rust
#![feature(const_transmute)]

const fn from_bits(v: u32) -> f32 {
    unsafe { std::mem::transmute(v) }
}

fn main() {}
```
Gives:
```
warning: transmute from a `u32` to a `f32`
 --> src/main.rs:4:14
  |
4 |     unsafe { std::mem::transmute(v) }
  |              ^^^^^^^^^^^^^^^^^^^^^^ help: consider using: `f32::from_bits(v)`
  |
  = note: `#[warn(clippy::transmute_int_to_float)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#transmute_int_to_float
```

But f32::from_bits can't be used in a const function yet.
